src/ssh/azext_ssh/custom.py

```diff
@@ -164,9 +164,16 @@ def _get_and_write_certificate(cmd, public_key_file, cert_file, ssh_client_folde
         # we used to use the username from the token but now we throw it away
         _, certificate = profile.get_msal_token(scopes, data)
     else:
-        credential, _, _ = profile.get_login_credentials(subscription_id=profile.get_subscription()["id"])
-        certificatedata = credential.get_token(*scopes, data=data)
-        certificate = certificatedata.token
+        if RUNNING_INSIDE_CLOUDSHELL:
```

Comment:
Running inside Cloud Shell doesn't mean this is a Cloud Shell account. User may manually run `az login` to override the Cloud Shell account.

CLI has more checks 

https://github.com/Azure/azure-cli/blob/d331edf76e5866cda1e6e9641d935256fc4f03ff/src/azure-cli-core/azure/cli/core/_profile.py#L319-L320

```py
        if in_cloud_console() and account[_USER_ENTITY].get(_CLOUD_SHELL_ID):
```